src/libostree/ostree-bootloader-syslinux.c

```diff
@@ -124,7 +133,19 @@ _ostree_bootloader_syslinux_write_config (OstreeBootloader  *bootloader,
     glnx_file_get_contents_utf8_at (self->sysroot->sysroot_fd, syslinux_config_path, NULL,
                                     cancellable, error);
   if (!config_contents)
-    return FALSE;
+    {
+      if (errno != ENOENT)
```

Comment:
Having `errno` be set after a function returns a `GError` is...somewhat undefined behavior.  It'd be better to do:
```
  if (g_error_matches (temp_error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
```
which is a pattern you can find all over the codebase (xref https://docs.rs/openat-ext/0.1.9/openat_ext/trait.OpenatDirExt.html#tymethod.open_file_optional )